File: sixtythreeebays_ui/src/services/CartService.js
import http from "../http-common"; 

class CartService {
  getAllCarts(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/cart/carts`, searchDTO);
  }

  get(cartId) {
    return this.getRequest(`/cart/${cartId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/cart?field=${matchData}`, null);
  }

  addCart(data) {
    return http.post("/cart/addCart", data);
  }

  update(data) {
  	return http.post("/cart/updateCart", data);
  }
  
  uploadImage(data,cartId) {
  	return http.postForm("/cart/uploadImage/"+cartId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new CartService();

File: src/main/java/com/sixtythreeebays/service/CartService.java
package com.sixtythreeebays.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.sixtythreeebays.domain.Cart;
import com.sixtythreeebays.dto.CartDTO;
import com.sixtythreeebays.dto.CartSearchDTO;
import com.sixtythreeebays.dto.CartPageDTO;
import com.sixtythreeebays.dto.CartConvertCriteriaDTO;
import com.sixtythreeebays.service.GenericService;
import com.sixtythreeebays.dto.common.RequestDTO;
import com.sixtythreeebays.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface CartService extends GenericService<Cart, Integer> {

	List<Cart> findAll();

	ResultDTO addCart(CartDTO cartDTO, RequestDTO requestDTO);

	ResultDTO updateCart(CartDTO cartDTO, RequestDTO requestDTO);

    Page<Cart> getAllCarts(Pageable pageable);

    Page<Cart> getAllCarts(Specification<Cart> spec, Pageable pageable);

	ResponseEntity<CartPageDTO> getCarts(CartSearchDTO cartSearchDTO);
	
	List<CartDTO> convertCartsToCartDTOs(List<Cart> carts, CartConvertCriteriaDTO convertCriteria);

	CartDTO getCartDTOById(Integer cartId);







}





